docs/custom-domain/registry-setup.md

```diff
@@ -0,0 +1,94 @@
+# Using Custom Domains with Azure Container Registry
+
+Every ACR is accessed using its login server. If you have a registry called `myregistry`, you access it using its default hostname, `myregistry.azurecr.io` (in Azure Public Cloud.) As a customer belonging to an organization, you may prefer to access your registry using a custom domain that is associated with your organization, for instance, `container-registry.contoso.com`.
+
+The following steps describe how you can achieve this.
+
+## Prerequisites
+- [Azure CLI](https://docs.microsoft.com/cli/azure/?view=azure-cli-latest), [curl](https://curl.haxx.se/) (or a similar tool)
+  - Consider using [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview)
+- A _premium_ Azure Container Registry. See [here](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-azure-cli) for instructions on how to create one.
+- Your custom domain names. The following two are required:
+  - custom registry domain to access the registry REST API: `container-registry.contoso.com` 
+  - custom data domain to access the registry content: `eastus-registry-data.contoso.com`
+    - Note that the custom data domain is region specific. For geo-replicated registries, each region should have its own custom data endpoint.
+
+  For each domain, you must prepare a single PEM formatted file containing the TLS private key and public certificate:
+  
+  ```
+  -----BEGIN PRIVATE KEY-----  
+  .....  
+  -----END PRIVATE KEY-----  
+  -----BEGIN CERTIFICATE-----  
+  .....  
+  -----END CERTIFICATE-----
+  ```
+  
+  For example, using [openssl](https://github.com/openssl/openssl):
+  - Create a self-signed public cert and private key
+     - `openssl req -nodes -x509 -newkey rsa:4096 -keyout container-registry.contoso.com.key.pem -out container-registry.contoso.com.cert.pem -days 365 -subj '/CN=container-registry.contoso.com/O=Contoso./C=US'`
+  - Create a single file containing both the public certificate and private key
+     - `cat container-registry.contoso.com.key.pem >> container-registry-contoso-com-pem`
+     - `cat container-registry.contoso.com.cert.pem >> container-registry-contoso-com-pem`
+  
+  Azure Key Vault allows you to [create certificates](https://docs.microsoft.com/azure/key-vault/certificate-scenarios) signed with a Certificate Authority.
+  
+## Prepare your existing registry
+We will enable two features on your registry that are currently in preview:
+- Data Endpoints:\
+  This feature provides a dedicated endpoint for downloading content from your registry. If you have a registry in East US, on enabling this feature, a data endpoint is automatically created for you: `eastus.data.myregistry.azurecr.io`
```

Comment:
This is not consistent with what is said above, with the dashes: eastus-data-myregistry.azurecr.io.

Reply:
It is not necessarily wrong, but should we recommend the customers to use a similar approach as we do?

Reply:
Also, if I have geo replication, are other data endpoints automatically created for me?

Reply:
Which line above this are you referring to?

Reply:
Line 13

Reply:
Line 13 provides an example for a custom data domain. The custom data domain could be anything, as chosen by a user.
This example is for a data endpoint that is ACR owned and automatically created for a registry when data endpoint is enabled.

Reply:
Also, for a custom owned domain called `contoso.com`, if more dots are prefixed to the name, that will also change the domain.